lib/manager/swift/extract.js

```diff
@@ -0,0 +1,346 @@
+const PACKAGE = 'package';
+const URL_KEY = 'urlKey';
+const STRING_LITERAL = 'stringLiteral';
+const COMMA = 'comma';
+const FROM = 'from';
+const RANGE_OP = 'rangeOp';
+const EXACT_VERSION = 'exactVersion';
+
+const searchLabels = {
+  wildcard: WILDCARD,
+  space: SPACE,
+  depsKeyword: DEPS,
+  colon: COLON,
+  beginSection: BEGIN_SECTION,
+  endSection: END_SECTION,
+  package: PACKAGE,
+  urlKey: URL_KEY,
+  stringLiteral: STRING_LITERAL,
+  comma: COMMA,
+  from: FROM,
+  rangeOp: RANGE_OP,
+  exactVersion: EXACT_VERSION,
+};
+
+function searchKeysForState(state) {
+  switch (state) {
+    case 'dependencies':
+      return [SPACE, COLON, WILDCARD];
+    case 'dependencies:':
+      return [SPACE, BEGIN_SECTION, WILDCARD];
+    case 'dependencies: [':
+      return [SPACE, PACKAGE, END_SECTION];
+    case '.package(':
+      return [SPACE, URL_KEY, PACKAGE, END_SECTION];
+    case '.package(url':
+      return [SPACE, COLON, PACKAGE, END_SECTION];
+    case '.package(url:':
+      return [SPACE, STRING_LITERAL, PACKAGE, END_SECTION];
+    case '.package(url: [depName]':
+      return [SPACE, COMMA, PACKAGE, END_SECTION];
+    case '.package(url: [depName],':
+      return [
+        SPACE,
+        FROM,
+        STRING_LITERAL,
+        RANGE_OP,
+        EXACT_VERSION,
+        PACKAGE,
+        END_SECTION,
+      ];
+    case '.package(url: [depName], .exact(':
+      return [SPACE, STRING_LITERAL, PACKAGE, END_SECTION];
+    case '.package(url: [depName], from':
+      return [SPACE, COLON, PACKAGE, END_SECTION];
+    case '.package(url: [depName], from:':
+      return [SPACE, STRING_LITERAL, PACKAGE, END_SECTION];
+    case '.package(url: [depName], [value]':
+      return [SPACE, RANGE_OP, PACKAGE, END_SECTION];
+    case '.package(url: [depName], [rangeFrom][rangeOp]':
+      return [SPACE, STRING_LITERAL, PACKAGE, END_SECTION];
+    default:
+      return [DEPS];
+  }
+}
+
+function getMatch(str, state) {
+  const keys = searchKeysForState(state);
+  let result = null;
+  for (let i = 0; i < keys.length; i += 1) {
+    const key = keys[i];
+    const regex = regExps[key];
+    const label = searchLabels[key];
+    const match = str.match(regex);
+    if (match) {
+      const idx = match.index;
+      const substr = match[0];
+      const len = substr.length;
+      if (idx === 0) {
+        return { idx, len, label, substr };
+      }
+      if (!result || idx < result.idx) {
+        result = { idx, len, label, substr };
+      }
+    }
+  }
+  return result;
+}
+
+function urlToDepName(url) {
+  try {
+    const { host, pathname } = new URL(url);
+    if (host === 'github.com' || host === 'gitlab.com') {
+      return pathname
+        .replace(/^\//, '')
+        .replace(/\.git$/, '')
+        .replace(/\/$/, '');
+    }
+    return url;
+  } catch (e) {
+    return null;
+  }
+}
+
+function extractPackageFile(content, packageFile = null) {
+  if (!content) return null;
+
+  const result = {
+    packageFile,
```

Comment:
I don't think we want to set this if `packageFile` is null. Or will null only happen in testing?

Reply:
Yes, it should be null only when testing